WISE-Community/WISE
# Added options to milestone report graphs

Authors can now customize title and bar color for milestone report graphs. Can also override the default possible score values for c-rater annotations (defaults are 1-5 for KI, 1-3 for other subscores).

To test:
- Graph title color:
  - In the authoring tool for a unit with a milestone report, add the `title-color` attribute to any `<milestone-report-graph>` element in a milestone report `template`. Value needs to be in the format of an rgb color value. For example: `title-color="rgb(219, 113, 28)"`.
  - Verify that the graph title matches the color you set when viewing the milestone report in the grading tool.
- Bar color:
  - In the authoring tool for a unit with a milestone report, add the `bar-color` attribute to any `<milestone-report-graph>` element in a milestone report `template`. Value needs to be in the format of an rgb color value. For example: `bar-color="rgb(219, 113, 28)"`.
  - Verify that the bars in the graph are gradations of the color you set when viewing the milestone report in the grading tool.
- Custom score categories:
  - To override the default possible scores for a c-rater subscore, add a `customScoreValues` object to a `report` entry in the `achievements` project json (in the authoring tool). Replace 'dci' in the following example with the subscore id you'd like to modify. The value should be an array of integers.
  - Make sure that the bar graph for the corresponding subscore id uses categories that match the custom score values set in the authoring and that the count for each category in the graph matches the number of students who received that score.
  - Example `achievements` json:
```
"achievements": {
  "isEnabled": true,
  "items": [
    ...
    "report": {
      "audience": [
        "teacher"
      ],
      "isEnabled": true,
      "customScoreValues": {
        "dci": [0,1,2]
       },
       "templates": [
         ...
```

Resolves #2250.
